client/client-lib/src/lib.rs

```diff
@@ -430,23 +431,51 @@ impl<T: AsRef<ClientConfig> + Clone> Client<T> {
 }
 
 impl Client<UserClientConfig> {
-    pub async fn fetch_gateway(&self) -> Result<LightningGateway> {
-        // fetch gateway from db
-        if let Some(gateway) = self
-            .context
+    pub async fn fetch_registered_gateways(&self) -> Result<Vec<LightningGateway>> {
+        let gateways = self.context.api.fetch_gateways().await?;
+        let current_active = self.fetch_active_gateway().await;
+
+        if current_active.is_some() {
+            // We have an active gateway in this client. We should update the list of registered gateways with that information.
+            Ok(gateways
+                .into_iter()
+                .map(|mut gateway| {
+                    if gateway.node_pub_key == current_active.clone().unwrap().node_pub_key {
+                        gateway.active = true;
+                    } else {
+                        gateway.active = false;
+                    }
+                    gateway
+                })
+                .collect())
+        } else {
+            Ok(gateways)
+        }
+    }
+    pub async fn fetch_active_gateway(&self) -> Option<LightningGateway> {
```

Comment:
Someday, maybe. But it's not very useful. Client needs to be online in order to do a payment so it can always ask for more gateway when they are needed. It would be useful if the client wanted to keep track of payment success / failure by gateway. But that's still far in the future.

Reply:
Agreed. I expect this to naturally bubble up again as we flesh out gateway management logic